source/dub/internal/vibecompat/core/file.d

```diff
@@ -156,8 +156,52 @@ private bool sameFile(NativePath a, NativePath b)
 	}
 }
 
+private bool isWritable(NativePath name)
+{
+	version (Windows)
+	{
+		import core.sys.windows.windows;
+
+		return (name.toNativeString.getAttributes & FILE_ATTRIBUTE_READONLY) == 0;
+	}
+	else version (Posix)
+	{
+		import core.sys.posix.sys.stat;
+
+		return (name.toNativeString.getAttributes & S_IWUSR) != 0;
+	}
+	else
+		static assert(false, "Needs implementation.");
+}
+
+private void makeWritable(NativePath name)
+{
+	makeWritable(name.toNativeString);
+}
+
+private void makeWritable(string name)
+{
+	version (Windows)
+	{
+		import core.sys.windows.windows;
+
+		name.setAttributes(name.getAttributes & ~FILE_ATTRIBUTE_READONLY);
```

Comment:
Can this operations fail (should it return a bool)?

Reply:
No, although setAttributes will throw if the file does not exist.

Reply:
So yes it can fail, but I don't think it needs to return `bool` as `copyFile` also may throw.

Reply:
Fail if not exists I think is fine to leave as is.